gtsam/gtsam.i

```diff
@@ -1068,6 +1068,23 @@ typedef gtsam::PinholeCamera<gtsam::Cal3DS2> PinholeCameraCal3DS2;
 typedef gtsam::PinholeCamera<gtsam::Cal3Unified> PinholeCameraCal3Unified;
 typedef gtsam::PinholeCamera<gtsam::Cal3Bundler> PinholeCameraCal3Bundler;
 
+template<T>
+class CameraSet {
+  CameraSet();
+
+  // common STL methods
+  size_t size() const;
+  bool empty() const;
+  void clear();
+
+  // structure specific methods
+  T at(size_t i) const;
+  void push_back(const T& cam);
+};
+
+typedef gtsam::CameraSet<gtsam::PinholeCameraCal3_S2> CameraSetCal3_S2;
+typedef gtsam::CameraSet<gtsam::PinholeCameraCal3Bundler> CameraSetCal3Bundler;
```

Comment:
Since `CameraSet` requires only one template, you can create a template list on line 1071 rather than defining multiple typedefs.

Reply:
Leaving the typedef there for shorter type name and to enforce uniformity in naming conventions

Reply:
rendered moot by adding preamble and CMake updates